src/lib/priceEncoder.ts

```diff
@@ -3,26 +3,12 @@ import bn from "bignumber.js";
 
 bn.config({ EXPONENTIAL_AT: 999999, DECIMAL_PLACES: 18 });
 
-function bigNumberToBig(val: BigNumber, decimals: number): bn {
-  return new bn(val.toString()).div(new bn(10).pow(decimals));
-}
-export function encodePriceSqrt(price: BigNumber): BigNumber {
+export function encodePriceSqrt(price: string | number): BigNumber {
   return BigNumber.from(
-    new bn(price.toString())
+    new bn(price)
```

Comment:
you might want to always instantiate bn with a string, otherwise it will throw if the number is too big

Reply:
We're passing in a price "ui" number, won't be too big :)